core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -205,29 +209,62 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             if (tags == null || tags.isEmpty()) {
                 continue;
             }
-            // latest tag is the first
+
+            // Iterate all imagestream tags and get the latest one by 'created' attribute
+            TagEvent latestTag = null;
+
             TAG_EVENT_LIST:
             for (NamedTagEventList list : tags) {
                 List<TagEvent> items = list.getItems();
```

Comment:
I would replace this code with:

```
List<TagEvent> items = list.getItems()!=null ? list.getItems() : Collections.EMPTY_LIST;
```
And remove if handling & continue with goto label altogether - if collection is empty, for loop would be noop.